unit_tests/cmake_linker.py

```diff
@@ -0,0 +1,36 @@
+import os
+from os.path import isfile, join
+
+ignore = ['CMakeLists.txt', 'cmake_linker.py']
+
+preamble = """include(FetchContent)
+FetchContent_Declare(
+	googletest
+        URL https://github.com/google/googletest/archive/refs/heads/main.zip
+        )
+
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+enable_testing()
+
+include(GoogleTest)\n"""
+
+def main():
+    cwd = os.path.abspath(os.getcwd())
+    testfiles = [f for f in os.listdir(cwd) if isfile(join(cwd, f)) and not f in ignore]
+
+    with open('CMakeLists.txt', 'w') as cmake:
+        cmake.write(preamble)
+
+        for test in testfiles:
+            cmake.write(f"\nadd_executable({test[:-3]} {test})\n")
+            cmake.write(f"target_link_libraries({test[:-3]} GTest::gtest_main)\n")
+            cmake.write(f"target_link_libraries({test[:-3]} iir::iir)\n")
+            cmake.write(f"gtest_discover_tests({test[:-3]})\n\n")
+
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Why have you mixed global variables with a main function. Imo stick to one or other

Reply:
didn't think it was an issue since the globals are just constants

Reply:
This is python though, not C++, and we are supposed to be doing good software engineer practices. No such thing a constant in python

Reply:
So are you proposing to just define them in the main function?